File: frontend/src/types/beta.ts
/**
 * Types for the beta solver API responses
 */

export interface BetaMove {
  /** Hold description e.g., "J4" */
  hold: string;
  /** Which hand to use */
  hand: "LH" | "RH";
}

export interface StartHand {
  /** Which hand (LH or RH) */
  hand: "LH" | "RH";
  /** Hold description e.g., "J4" */
  description: string;
}

export interface BetaResponse {
  /** Starting hand positions */
  startHands: StartHand[];
  /** Sequence of moves */
  moves: BetaMove[];
}

File: beta-solver/BetaSolver/BetaSolver.Core/Solver/Beta.cs
using BetaSolver.Core.Models;

namespace BetaSolver.Core.Solver;

/// <summary>
/// Represents the result of solving for the optimal beta sequence.
/// </summary>
/// <param name="Moves">The ordered list of scored moves in the sequence with full spatial context</param>
/// <param name="StartHands">Maps each hand to the starting hold it is on</param>
public sealed record Beta(
    IReadOnlyList<Move> Moves,
    IReadOnlyDictionary<Hand, Hold> StartHands)
{
    /// <summary>
    /// Creates an empty result with no moves.
    /// </summary>
    public static Beta Empty => new([], new Dictionary<Hand, Hold>());
}
